pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -164,6 +164,7 @@ def _copy_trainer_model_properties(self, model):
 
         for m in [model, ref_model]:
             m.trainer = proxy(self.trainer)
+            # Remove next line in v1.8
```

Comment:
Be more specific about use_amp in case the code here changes between now and v1.8